components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt

```diff
@@ -259,7 +269,7 @@ class AddonsManagerAdapter(
 
         addons.forEach { addon ->
             when {
-                addon.inUnsupportedSection() -> unsupportedAddons.add(addon)
+                addon.inUnsupportedSection() && !unsupportedAddons.contains(addon) -> unsupportedAddons.add(addon)
```

Comment:
Why do we need that additional guard? How come we add the same addon twice to unsupported addons? If we need to guard for this I would change this to a `Set`,

Reply:
As [unsupportedAddons](https://github.com/mozilla-mobile/android-components/blob/b25f0bf55a32888632be7186796342c1adbe4e0f/components/feature/addons/src/main/java/mozilla/components/feature/addons/ui/AddonsManagerAdapter.kt#L62) is global,  calling `createListWithSections` causes the side effect of adding multiple times to `unsupportedAddons`. Yes a `Set` could be a good alternative!